### PR TITLE
Added ibus RX support to recieve channel data from ibus transmitters.

### PR DESCRIPTION
**Pull request by PraiseSatan**
I have added a RCProtocol for ibus.
This will allow ibus based RX to be used without any adaptors.
This is only one way, with it only taking data from the RX, not sending any telemetry data back.

#9419

### CODE REVIEW

**Comment by magicrub** [comment]
Mind fixing this PR so it doesn't have a merge conflict?

**Comment by PraiseSatan** [comment]
I have had a look at how to fix the PR and am unsure exactly what to do as I am not used to using Github and am having difficulties.

The conflicting file is the compiled firmware for the IO processor, which would need to be recompiled to reflect the changes in both versions.
How can I just remove that file from the pull request?
Or would I need to get all of master and recompile?

**Comment by tridge** [comment]
@PraiseSatan I've fixed up the merge conflict for you

File: libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/*
+  IBUS decoder, based upon SBUS decoder which was based on src/modules/px4iofirmware/sbus.c from PX4Firmware
+  modified for use in AP_HAL_* by Andrew Tridgell, modified for decoding IBUS.
+ */
+/****************************************************************************
+ *
+ *   Copyright (c) 2012-2014 PX4 Development Team. All rights reserved.
```
Comment (tridge) [inline libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp]: Is it really based on work from PX4? It doesn't look like it. I think you could remove this.

File: libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Code by Andrew Tridgell and Siddharth Bharat Purohit
```
Comment (tridge) [inline libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp]: remove these incorrect author lines

File: libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp (hunk 1 above)
Comment (PraiseSatan) [reply libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp]: Indirectly yes.
I used the code for SBUS from here as a template to make sure things were set up as per the rest of ardupilot.
I didn't want to cause any copyright issues for lack of attribution so I thought it best to leave the copyright text mostly the same.

**Comment by PraiseSatan** [comment]
> Thanks for this work!
> I would like to see IBUS added to our test suite here:
> https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_RCProtocol/examples/RCProtocolTest/RCProtocolTest.cpp#L186
> can you add it?

Does this need any particular input value, or can it be any valid input with the corresponding output?
I have assumed the latter.

File: libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp (hunk 2 above)
Comment (PraiseSatan) [reply libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp]: I have removed these lines now.

**Comment by tridge** [comment]
Thanks! I've re-squashed it and its ready to go in when CI passes

**Pull request merged**